util/random.go

```diff
@@ -1,17 +1,56 @@
 package util
 
-import "math/rand"
+import (
+	cryptoRand "crypto/rand"
+	mathRand "math/rand"
+)
 
 // RandomString generates a random string of length n. Before calling this function, you should call
 // rand.Seed() to initialize the default source.
 //
 // Found at https://ispycode.com/Blog/golang/2016-10/How-to-generate-a-random-string-of-a-fixed-length
+//  - Adapted for more characters
 func RandomString(n int) string {
-	var letterRunes = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+	var letterRunes = []rune("0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz")
 
 	b := make([]rune, n)
 	for i := range b {
-		b[i] = letterRunes[rand.Intn(len(letterRunes))]
+		b[i] = letterRunes[mathRand.Intn(len(letterRunes))]
 	}
 	return string(b)
 }
+
+// CryptoRandomBytes returns securely generated random bytes.
+// It will return an error if the system's secure random
+// number generator fails to function correctly, in which
+// case the caller should not continue.
+//
+// Found at https://gist.github.com/dopey/c69559607800d2f2f90b1b1ed4e550fb
+func CryptoRandomBytes(n int) ([]byte, error) {
+	b := make([]byte, n)
+	_, err := cryptoRand.Read(b)
+	// Note that err == nil only if we read len(b) bytes.
+	if err != nil {
+		return nil, err
+	}
+
+	return b, nil
+}
+
+// CryptoRandomString returns a securely generated random string.
+// It will return an error if the system's secure random
+// number generator fails to function correctly, in which
+// case the caller should not continue.
+//
+// Found at https://gist.github.com/dopey/c69559607800d2f2f90b1b1ed4e550fb
+func CryptoRandomString(n int) (string, error) {
+	const letters = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-"
```

Comment:
I would prefer to (a) not use ASCII-sequential chars and (b) put special characters in here.

Suggest something like (which is different than the above random letters):

```
const letters = "gl9z2p1a%8c4Er3@XFyLvGf!ZMHubAYeTdUB$&nV6o#jRkWmC)qIPtS*QJ0h7(^K5OwDiNxs"
```

Where I randomized the order of the chars (simply via this command):

```
echo '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz!@#$%^&*()' | fold -w1 | shuf | tr -d '\n' ;echo
```

Reply:
I added the special characters. Not randomizing the order, though.